crates/utils/re_arrow_combinators/src/error.rs

```diff
@@ -34,6 +34,9 @@ pub enum Error {
         struct_fields: Vec<String>,
     },
 
+    #[error("Unexpected value: expected {expected}, got {actual}")]
+    UnexpectedValue { expected: String, actual: String },
```

Comment:
Nit: The `expected` could probably be a list of `&'static [&'static str]`. That way we would not need the `or` in the message nor the allocation.